Realm/ObjectServerTests/RLMObjectServerTests.m

```diff
@@ -735,31 +739,37 @@ - (void)testStreamingUploadNotifier {
     __block NSInteger callCount = 0;
     __block NSUInteger transferred = 0;
     __block NSUInteger transferrable = 0;
-    dispatch_queue_t queue = dispatch_get_global_queue(QOS_CLASS_USER_INTERACTIVE, 0);
     // Open the Realm
     RLMRealm *realm = [self openRealmForURL:url user:user];
     NSLock *lock = [[NSLock alloc] init];
     // Register a notifier.
     RLMSyncSession *session = [user sessionForURL:url];
     XCTAssertNotNil(session);
-    RLMProgressNotificationToken *token = [session addProgressNotificationBlock:^(NSUInteger xfr, NSUInteger xfb) {
-        [lock lockBeforeDate:[NSDate distantFuture]];
-        // Make sure the values are increasing, and update our stored copies.
-        XCTAssert(xfr >= transferred);
-        XCTAssert(xfb >= transferrable);
-        transferred = xfr;
-        transferrable = xfb;
-        callCount++;
-        [lock unlock];
-    } direction:RLMSyncNotifierDirectionUpload mode:RLMSyncNotifierModeAlwaysReportLatest queue:queue];
+    __block RLMProgressNotificationToken *token;
+    XCTestExpectation *ex = [self expectationWithDescription:@"token expectation"];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        token = [session addProgressNotificationBlock:^(NSUInteger xfr, NSUInteger xfb) {
+            [lock lockBeforeDate:[NSDate distantFuture]];
```

Comment:
Then again, this block should be invoked on the main thread so the locking looks to be unnecessary.